File: round1_full_pack_optionA_solutions/M2_Longest_Substring_No_Repeat/starter_code/solution.py
import sys
s=sys.stdin.readline().strip()
last={}
l=0; best=0
for r,ch in enumerate(s):
    if ch in last and last[ch] > l:  
        l = last[ch]                
    last[ch]=r
    best=max(best, r-l)             
print(best)

File: round1_full_pack_optionA_solutions/M2_Longest_Substring_No_Repeat/starter_code/solution.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
  string s; getline(cin,s);
  vector<int> last(256,-1);
  int l=0,best=0;
  for(int r=0;r<(int)s.size();r++){
    unsigned char c=s[r];
    if(last[c]>=l) l=last[c];     
    last[c]=r;
    best=max(best,r-l);           
  }
  cout<<best<<"\n";
}
